core/src/main/java/zipkin/reporter/AsyncReporter.java

```diff
@@ -244,13 +246,14 @@ void flush(BufferNextMessage bundler) {
 
     @Override
     public void close() {
-      closed.set(true);
+      if (!closed.compareAndSet(false, true)) return; // already closed
       try {
-        if (!close.await(messageTimeoutNanos, TimeUnit.NANOSECONDS)) {
-          logger.warning("Timed out waiting for close");
+        // wait a second for in-flight spans to send
+        if (!close.await(1, TimeUnit.SECONDS)) {
```

Comment:
Depending on the size of the buffer I'm not sure this would be enough time. Maybe it should be configurable aswell?